### PR TITLE
Changes to support Destination FQDNs in Network Rule and Translated FQDN in NAT Rule for Azure Firewall

### PR DESCRIPTION
**Pull request by tejasshah7**
## Description

Changes to support Destination FQDNs in Network Rule and Translated FQDN

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/Network/Network/AzureFirewall/NetworkRule/NewAzureFirewallNetworkRuleCommand.cs [hunk 1]
```diff
@@ -43,11 +44,17 @@ public class NewAzureFirewallNetworkRuleCommand : NetworkBaseCmdlet
...
+            Mandatory = false,
             HelpMessage = "The destination addresses of the rule")]
         [ValidateNotNullOrEmpty]
         public string[] DestinationAddress { get; set; }
 
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The destination FQDNs of the rule")]
+        [ValidateNotNullOrEmpty]
+        public string[] DestinationFqdns { get; set; }
```
Comment (number213) [inline src/Network/Network/AzureFirewall/NetworkRule/NewAzureFirewallNetworkRuleCommand.cs]: Cmdlet parameters should be in singular form

Comment (tejasshah7) [reply src/Network/Network/AzureFirewall/NetworkRule/NewAzureFirewallNetworkRuleCommand.cs]: Ok I will change this and commit along with test recordings.

**Comment by number213** [comment]
Closing in favor of https://github.com/Azure/azure-powershell/pull/10388/

**Pull request closed without merging**